workflow/util/util.go

```diff
@@ -848,6 +857,20 @@ func FormulateRetryWorkflow(ctx context.Context, wf *wfv1.Workflow, restartSucce
 			version := GetWorkflowPodNameVersion(wf)
 			podName := PodName(wf.Name, node.Name, templateName, node.ID, version)
 			podsToDelete = append(podsToDelete, podName)
+
+			deletedNodes[node.ID] = true
+
+			descendantNodeIDs := getDescendantNodeIDs(wf, node)
+			for _, descendantNodeID := range descendantNodeIDs {
+				deletedNodes[descendantNodeID] = true
+				descendantNode := wf.Status.Nodes[descendantNodeID]
+				if descendantNode.Type == wfv1.NodeTypePod {
+					templateName := getTemplateFromNode(descendantNode)
+					version := GetWorkflowPodNameVersion(wf)
+					podName := PodName(wf.Name, descendantNode.Name, templateName, descendantNode.ID, version)
+					podsToDelete = append(podsToDelete, podName)
```

Comment:
I suspect there will be redundant pod names here.

Reply:
done